### DISCUSSION TITLE
Review of Speech input

### DISCUSSION THREAD

**Discussion by AlexeyMatveev686** [ONLYOFFICE/onlyoffice.github.io#249] [Polls]
<img src="https://raw.githubusercontent.com/ONLYOFFICE/onlyoffice.github.io/master/discussions-icons/speechrecognition.svg">

**Comment by adocampo** [comment]
Doesn't seems to work at all. Just a pop up appears, clicked on it, microphone icon changed the color and the message "Speak now" appears, but then nothing, nada.

**Reply by AlexeyMatveev686** [reply]
Hello. Give a more information, please. Which version of plugin and editors do you use? What is your OS and browser? Did you try to use it into desktop or document server?

**Reply by adocampo** [reply]
Sure. 
 - ONLYOFFICE Desktop Editors version 7.5.1.23
 - Plugin version 1.0.1.
 - OS: Arch Linux
 - Browser: N/A, as I'm using the Desktop edition of OnlyOffice.

**Reply by AlexeyMatveev686** [reply]
Thanks for information. But unfortunally this plugin doesn't work in desktop editors. In the feature we will plan to fix it. You can watch the "[known issues](https://github.com/ONLYOFFICE/onlyoffice.github.io/tree/master/sdkjs-plugins/content/speechrecognition#known-issues)" item into the README of this plugin (we will change this file when we fix the problem).

**Reply by adocampo** [reply]
Oh, I did install it via plugin management system under the editor itself and I didn't figure it couldn't be compatible. I will try to install the dockerized version of OnlyOffice and try it with firefox, as it seems it's compatible after modifying a couple of flags. Just I saw a possible typo in the documentation:
![image](https://github.com/ONLYOFFICE/onlyoffice.github.io/assets/2266612/f387dcd2-7c09-47e5-8474-7dfb11c5e566)
If the flags are switched "on" by default, should I disable them or if I need to "enable" them, then, they are switched "off" by default?

**Reply by AlexeyMatveev686** [reply]
Thank you for your tips. It's our typo. This flags are swithced "off" by default. We will fix it soon into README.

**Reply by freedroid94** [reply]
Minor, but very irritating, gripe of mine: there is no such thing as "Brasileira" as a language, it's Portuguese

**Reply by Misokados** [reply]
Hello @freedroid94 
It seems to be fixed:
<img width="149" height="201" alt="image" src="https://github.com/user-attachments/assets/d3cbf16d-d61d-42d8-8d8e-d7dc242db9c9" />

**Comment by adocampo** [comment]
Thank you, on Chrome browser seems to work fine. On firefox, though, it doesn't work even with the tunning options enabled.
> Error with connection to recognition service of this browser, please use the Chrome browser

Document Editor: 7.5.1.23
Firefox version: 121.0.1 
Docker version 24.0.7, build afdd53b4e3
Docker Compose version 2.24.1

**Reply by AlexeyMatveev686** [reply]
Thanks for your report. I've checked the oficial [page ](https://developer.mozilla.org/en-US/docs/Web/API/Web_Speech_API/Using_the_Web_Speech_API)with example of Web Speech Api, and tried to use their [example](https://mdn.github.io/dom-examples/web-speech-api/speech-color-changer/). This page gave me the same error as our plugin. I guess that problem could be into Firefox browser engine, because in Chrome it works fine.

**Comment by oqolopo** [comment]
The next message is still in wherever you do!
An embedded page at onlyoffice.github.io says
Permission to use microphone is blocked. To change, go to chrome://settings/content/microphone

**Reply by Misokados** [reply]
Hello,
please describe the situation as detailed as possible. You can record a video file demonstrating the issue.

**Comment by liar666** [comment]
Using FF. Clicking on button asks to allow Mic access. Then says it needs Chrome. 

Aren't access to Mic/etc. supposed to be API calls, thus universal & not requiring a specific browser/OS/... ????

**Reply by Misokados** [reply]
Hello,
https://github.com/ONLYOFFICE/onlyoffice.github.io/blob/master/sdkjs-plugins/content/speechrecognition/README.md#:~:text=This%20plugin%20works%20in%20Google%20Chrome%20only.%20To
_To use it in FireFox, enable recognition via media.webspeech.recognition.enable and media.webspeech.recognition.force_enable flags in about:config synthesis is switched on by default_
